tests/linalg/inverse/test_pseudo.py

```diff
@@ -0,0 +1,25 @@
+from cola.linalg.inverse.pseudo import pseudo
+from cola.linalg.algorithm_base import Auto
+from cola.ops import LinearOperator
+from cola.utils.test_utils import parametrize, relative_error
+from cola.backends import all_backends
+from nonsquare_operator_market import get_test_operator
+
+
+@parametrize(all_backends, ["float64"], ["psd_diagonal", "square_dense"])
+def test_pseudo(backend, precision, op_name):
+    operator = get_test_operator(backend, precision, op_name)
+    tol = 1e-5
+    A, dtype, xnp = operator, operator.dtype, operator.xnp
+    A2 = LinearOperator(A.dtype, A.shape, A._matmat)
+    Ainv = pseudo(A, Auto(tol=tol))
+    Ainv2 = pseudo(A2, Auto(tol=tol))
+    Ainv3 = pseudo(A2, Auto(tol=tol, method="iterative"))
```

Comment:
this is the kind of calling we want to avoid

Reply:
When you make this statement you are implying that it is self-evident why this is a problem. Would you mind elaborating more. Are you unhappy with the use of `pseudo`, or `Auto` receiving `method.`